integration-tests/src/nft_marketplace.rs

```diff
@@ -135,3 +138,102 @@ fn create_class_should_fail_when_relay_chain_location_not_registered() {
 		);
 	});
 }
+
+#[test_case(ClassType::Marketplace, true ; "marketplace class")]
+#[test_case(ClassType::LiquidityMining, false ; "liquidity mining class")]
+#[test_case(ClassType::Redeemable, false ; "redeemable class")]
+#[test_case(ClassType::Auction, false ; "auction class")]
+#[test_case(ClassType::HydraHeads, false ; "hydra heads class")]
+fn nft_permissions_should_allow_marketplace_class_creation(class_type: ClassType, is_allowed: bool) {
```

Comment:
Great job with the tests!

Just one thing as stated in one of my other comments, we could make this even more readable by introducing two consts:
```
const ALLOW: bool = true;
const NOT_ALLOW: bool = false;
```

And using these in the `test_case` attributes. By doing this, we could remove the magic booleans. But as you want, I approve the PR anyway. ;)